test/e2e/specs/i18n/i18n__editor.spec.ts

```diff
@@ -0,0 +1,71 @@
+ *
+ * This suite iterates over a set of locales and their expected "Add title" translations, and for each:
+ * - Authenticates as a test account (using/saving cookies for efficiency).
+ * - Updates the account's locale setting using the REST API.
+ * - Visits the editor page.
+ * - Asserts that the editor's title input displays the correct localized placeholder.
+ *
+ * Tests are run serially to avoid conflicts from changing the shared account's locale.
+ * Tests are restricted to run only in the Chrome browser.
+ *
+ * @file I18N: Editor E2E tests for localized editor content.
+ * @tag @i18n
+ * @see Translation
+ */
+import { RestAPIClient } from '@automattic/calypso-e2e';
+import { test, expect } from '../../lib/pw_base';
+import { Translation } from '../../lib/types_shared';
+
+const translationsToTest: Array< Translation > = [
+	{ locale: 'ar', addTitle: 'أضف عنوانًا' },
+	{ locale: 'de', addTitle: 'Titel hier eingeben' },
+	{ locale: 'en', addTitle: 'Add title' },
+	{ locale: 'es', addTitle: 'Añadir título' },
+	{ locale: 'fr', addTitle: 'Ajout de titre' },
+	{ locale: 'he', addTitle: 'הוספת כותרת' },
+	{ locale: 'id', addTitle: 'Tambahkan judul' },
+	{ locale: 'it', addTitle: 'Aggiungi un titolo' },
+	{ locale: 'ja', addTitle: 'タイトルを追加' },
+	{ locale: 'ko', addTitle: '제목 추가' },
+	{ locale: 'nl', addTitle: 'Titel toevoegen' },
+	{ locale: 'pt-br', addTitle: 'Adicionar título' },
+	{ locale: 'ru', addTitle: 'Добавить заголовок' },
+	{ locale: 'sv', addTitle: 'Lägg till rubrik' },
+	{ locale: 'tr', addTitle: 'Başlık ekle' },
+	{ locale: 'zh-cn', addTitle: '添加标题' },
+	{ locale: 'zh-tw', addTitle: '新增標題' },
+];
+
+test.describe( 'I18N: Editor', { tag: '@i18n' }, () => {
+	test.describe.configure( { mode: 'serial' } ); // Since all tests use the same account which changes its locale, they should not be run in parallel
+	for ( const translation of translationsToTest ) {
+		test( `As an i18n visitor using '${ translation.locale }' as my locale I can see localised editor content`, async ( {
+			page,
+			accounti18n,
+			pageEditor,
+		}, workerInfo ) => {
+			test.skip( workerInfo.project.name !== 'chrome', 'The i18n editor tests only run in Chrome' );
+
+			await test.step( `Given I am authenticated as '${ accounti18n.accountName }'`, async function () {
+				await accounti18n.authenticate( page );
+			} );
+
+			await test.step( `And I update my locale settings to ${ translation.locale }`, async function () {
+				const clientRestAPI = new RestAPIClient( accounti18n.credentials );
+				await clientRestAPI.setMySettings( { language: translation.locale } );
+				await page.reload();
+			} );
+
+			await test.step( 'When I visit the editor page', async function () {
+				await pageEditor.visit( 'post' );
+			} );
+
+			await test.step( 'Then I can see the following see the correct translations', async function () {
+				await expect( page.getByRole( 'heading', { level: 1, name: translation.addTitle } ) )
```

Comment:
@aduth  actually I now realised why I didn't use this approach as when it fails, say the translation fails, it only tells you it's not visible and also takes longer to fail and it keeps looking for the old translation.

What do you think of this approach? I feel like it's the best of both in that it uses a role, but also asserts the accessible text is correct and gives a diff when it fails.

`await expect( page.getByRole( 'textbox' ) ).toHaveAccessibleName( translation.addTitle );`

Example failure:

```
    Locator: getByRole('textbox')
    Expected string: "Add Title"
    Received string: "Add title"
 ```

Reply:
> What do you think of this approach?

Does that assume that the page would only have one text field, or at least that the first one it matches is the title field? I'd worry those are pretty fragile assumptions.

I don't know that I follow what you mean by "keeps looking for the old translation", but to the point of unclear error messages, we could specify a [custom failure message](https://playwright.dev/docs/test-assertions#custom-expect-message)?